test/unit/lib/plugins/aws/package/compile/functions.test.js

```diff
@@ -2369,22 +1444,71 @@ describe('AwsCompileFunctions', () => {
 describe('lib/plugins/aws/package/compile/functions/index.test.js', () => {
   describe('Provider properties', () => {
     let cfResources;
+    let cfOutputs;
     let naming;
     let serviceConfig;
-    let iamRolePolicyStatements;
 
     before(async () => {
+      const serviceArtifactPath = path.join(process.cwd(), 'testServiceArtifact.zip');
+      await fs.promises.writeFile(serviceArtifactPath, 'packageContent');
+      const functionArtifactPath = path.join(process.cwd(), 'testFunctionArtifact.zip');
+      await fs.promises.writeFile(functionArtifactPath, 'packageContent');
+
       const { awsNaming, cfTemplate, fixtureData } = await runServerless({
         fixture: 'function',
         cliArgs: ['package'],
         configExt: {
+          service: {
+            name: 'testService',
```

Comment:
Service name is set by fixture handler, let's not configure this property

Reply:
At first I didn't want to configure it, but it's a required property here and if it's omitted, it's resulting in a configuration error: `ServerlessError: Configuration error at 'service': should have required property 'name'`. I've changed the name to `service` as in fixtures, but I'm open to better suggestions :+1:

Reply:
Ok, I see.

Situation is that fixture generator, generates service name, but it [unconditionally assigns it to `service` as string property](https://github.com/serverless/test/blob/a1575a97f44e3904c5db0a5e4a821af6624dfcdd/get-fixtures-engine.js#L106). This makes sense, as technically we deprecated _object_ representation of `service`, so we should not embrace the object format anymore

Still in this case we want to test _deprecated_ notation and for that we need an _object_ form of `service`.

I'd say keep it as you've put, but let's add a `TODO` comment over `name` that whole `service`object configuration should be removed when removing the support for `service.awsKmsKeyArn`

(otherwise setting `service.name` is confusing, as it's set automatically to unique value for each setup fixture)

Reply:
Thanks for a detailed explanation :bow: I've added a `TODO` as suggested